ops/model.py

```diff
@@ -3640,12 +3642,33 @@ def secret_set(
         args = [id]
         if label is not None:
             args.extend(['--label', label])
+            self._secret_cache[id]['label'] = label
+        elif 'label' in self._secret_cache[id]:
+            args.extend(['--label', self._secret_cache[id]['label']])
         if description is not None:
             args.extend(['--description', description])
+            self._secret_cache[id]['description'] = description
+        elif 'description' in self._secret_cache[id]:
+            args.extend(['--description', self._secret_cache[id]['description']])
         if expire is not None:
             args.extend(['--expire', expire.isoformat()])
+            self._secret_cache[id]['expire'] = expire
+        elif 'expire' in self._secret_cache[id]:
+            args.extend(['--expire', self._secret_cache[id]['expire'].isoformat()])
         if rotate is not None:
             args += ['--rotate', rotate.value]
+            self._secret_cache[id]['rotate'] = rotate
+        elif 'rotate' in self._secret_cache[id]:
+            args += ['--rotate', self._secret_cache[id]['rotate'].value]
+
+        if content is not None:
+            # Don't store the actual secret content in memory, but put a sentinel there to indicate
+            # that the content has been set during this hook.
+            self._secret_cache[id]['content'] = object()
```

Comment:
I wonder: does this matter, especially given that hooks are short-lived (which this cache depends on, right)? We already cache the content on a `Secret` object -- though I guess the lifetime of those are shorter-lived?

Reply:
Hooks are typically short-lived, and this cache indeed is just in memory so only lives as long as the model backend does.

With the `Secret` object, we cache when you *get* the (tracked) content, but we don't when you *set* the (new latest) content. I also feel like it's also more predictable that the content might be cached on the `Secret` object, rather than on the model backend (which might be processing multiple events, for example).

My preference is to not expand the caching of secret content (this would be doubly: both new content on set, and also the model backend rather than the secret object).

Reply:
Discussed during daily and there were no objections to avoiding additional caching of sensitive content.